azurelinuxagent/common/dhcp.py

```diff
@@ -86,9 +86,8 @@ def wireserver_route_exists(self):
         route_exists = False
         logger.info("Test for route to {0}".format(KNOWN_WIRESERVER_IP))
         try:
-            route_file = '/proc/net/route'
-            if os.path.exists(route_file) and \
-                    KNOWN_WIRESERVER_IP_ENTRY in open(route_file).read():
+            route_table = self.osutil.read_route_table()
+            if any([(KNOWN_WIRESERVER_IP_ENTRY in route) for route in route_table]):
```

Comment:
In the original code there was a check for exists(route_file); if it was false, we would output a warning.
In the new code this would be reported as an error.

Reply:
I'm not sure how to work around this. read_route_table will log an error by default and then return an empty list if /proc/net/route doesn't exist.

The only thing it guarantees is not to raise an exception.

Im not sure changing that function would be appropriate. I do think using that function would be appropriate since checking route is clearly a platform specific task.

Reply:
I also do think that this is an edge case and not really an issue as it will only log an error on systems without /proc/net/route (non Linux systems) which have not overridden that function with their own osutil.

Reply:
ok